app/models.py

```diff
@@ -429,14 +429,33 @@ def serialize(self, data=None):
             "frameworkSlug": self.framework.slug,
             "declaration": self.declaration,
             "onFramework": self.on_framework,
-            "agreementReturned": bool(agreement_returned_at),
-            "agreementReturnedAt": agreement_returned_at,
-            "agreementDetails": self.agreement_details,
-            "countersigned": bool(countersigned_at),
-            "countersignedAt": countersigned_at,
             "agreedVariations": agreed_variations,
         }, **(data or {}))
 
+        if self.framework_agreements:
+            agreement = self.framework_agreements[0]
```

Comment:
Sorry, I forgot to mention in the pull request that we are planning on getting everything working under the current assumption that the relationship will be one to one. We were trying to keep the refactoring steps small and will then need to change this as a later step.